### ISSUE TITLE
Making new release for GAP 4.9 beta

### ISSUE THREAD

**Issue by olexandr-konovalov** [hulpke/transgrp#8]
@hulpke we would like to make a complete transition from `small`, `prim` and `trans` to SmallGrp, PrimGrp and TransGrp packages before releasing GAP 4.9 beta, i.e. GAP 4.9.0. How far is `TransGrp` from being released? Shall I try to pick up the Summer 2016 version from http://www.math.colostate.edu/~hulpke/transgrp/, or wait for version 2.0? Would you be interested in hosting it on GitHub and using @fingolfin's ReleaseTools and GitHubPagesForGAP to create a website and automated releases?

**Comment by hulpke** [comment]
Yes, I think it would be a good idea to avoid keeping the trans directory and package in parallel. As the groups are used extensively in examples and tests it would have to be a required package as ctbllib, tom and gapdoc are.

As far as I recall, the Summer '16 version of the package provides all functionality that GAP 4.8 has and so should be included. I do not have a new release at hand and the 4.9 release of GAP should not wait for any package, at least not this one.

However if there are concrete problems that something does not work (beyond ``it would be nice to have a fancier README file) please let me know and I will of course make a release in which these are fixed.

**Comment by olexandr-konovalov** [comment]
Yes, as soon as the package will be picked up by the package update system and will work, the transition may look similar to the one I suggest at https://github.com/gap-system/gap/pull/1650 (if you can review that PR, it would be great!). 

I will be happy to try to pick it up and see how it behaves in tests in Jenkins, and then let you know if there is anything. `git log trans` shows some changes in the `trans` directory made after summer 2016, so one may need to port them to TransGrp - either by releasing new version 2.0 which will include also new groups, or by making a bugfix release.

Just in case,  GAPDoc is needed to run GAP, but CTblLib and TomLib are not.

**Comment by olexandr-konovalov** [comment]
Scheduled for pick up in https://github.com/gap-system/gap-distribution/commit/f838da5922e109a492ae3e881b80671e34e8f6e5 - waiting to see what happens.

**Comment by olexandr-konovalov** [comment]
Picked up, PackageInfo.g passed validation:
```
* transgrp - new version 1.0 discovered!!!
  ============================================
  Getting new archives from 
  http://www.math.colostate.edu/~hulpke/transgrp/transgrp1[ ".tar.gz" ]
  unpacking transgrp1.tar.gz
  Validation of the info file successful!
  Package transgrp 1.0 from 30/06/2016 has dependencies:
  * GAP >=4.9
  finding text files  . . .

=====================text files==========================
transgrp/init.g
transgrp/LICENSE
transgrp/PackageInfo.g
transgrp/read.g
transgrp/README
transgrp/data/trans.gd
transgrp/data/trans.gi
transgrp/data/trans.grp
transgrp/data/trans10.grp
transgrp/data/trans11.grp
transgrp/data/trans12.grp
transgrp/data/trans13.grp
transgrp/data/trans14.grp
transgrp/data/trans15.grp
transgrp/data/trans16.grp
transgrp/data/trans16a.grp
transgrp/data/trans16b.grp
transgrp/data/trans16c.grp
transgrp/data/trans16d.grp
transgrp/data/trans16e.grp
transgrp/data/trans16f.grp
transgrp/data/trans16g.grp
transgrp/data/trans17.grp
transgrp/data/trans18.grp
transgrp/data/trans18a.grp
transgrp/data/trans18b.grp
transgrp/data/trans18c.grp
transgrp/data/trans18d.grp
transgrp/data/trans19.grp
transgrp/data/trans20.grp
transgrp/data/trans20a.grp
transgrp/data/trans20b.grp
transgrp/data/trans20c.grp
transgrp/data/trans20d.grp
transgrp/data/trans21.grp
transgrp/data/trans21a.grp
transgrp/data/trans22.grp
transgrp/data/trans22a.grp
transgrp/data/trans23.grp
transgrp/data/trans24.grp
transgrp/data/trans24a.grp
transgrp/data/trans24aa.grp
transgrp/data/trans24ab.grp
transgrp/data/trans24ac.grp
transgrp/data/trans24ad.grp
transgrp/data/trans24ae.grp
transgrp/data/trans24af.grp
transgrp/data/trans24ag.grp
transgrp/data/trans24ah.grp
transgrp/data/trans24ai.grp
transgrp/data/trans24aj.grp
transgrp/data/trans24ak.grp
transgrp/data/trans24al.grp
transgrp/data/trans24am.grp
transgrp/data/trans24an.grp
transgrp/data/trans24ao.grp
transgrp/data/trans24ap.grp
transgrp/data/trans24aq.grp
transgrp/data/trans24ar.grp
transgrp/data/trans24as.grp
transgrp/data/trans24at.grp
transgrp/data/trans24au.grp
transgrp/data/trans24av.grp
transgrp/data/trans24aw.grp
transgrp/data/trans24ax.grp
transgrp/data/trans24ay.grp
transgrp/data/trans24az.grp
transgrp/data/trans24b.grp
transgrp/data/trans24ba.grp
transgrp/data/trans24bb.grp
transgrp/data/trans24bc.grp
transgrp/data/trans24bd.grp
transgrp/data/trans24be.grp
transgrp/data/trans24bf.grp
transgrp/data/trans24bg.grp
transgrp/data/trans24bh.grp
transgrp/data/trans24bi.grp
transgrp/data/trans24bj.grp
transgrp/data/trans24bk.grp
transgrp/data/trans24bl.grp
transgrp/data/trans24bm.grp
transgrp/data/trans24bn.grp
transgrp/data/trans24bo.grp
transgrp/data/trans24bp.grp
transgrp/data/trans24bq.grp
transgrp/data/trans24br.grp
transgrp/data/trans24bs.grp
transgrp/data/trans24bt.grp
transgrp/data/trans24bu.grp
transgrp/data/trans24bv.grp
transgrp/data/trans24bw.grp
transgrp/data/trans24bx.grp
transgrp/data/trans24by.grp
transgrp/data/trans24bz.grp
transgrp/data/trans24c.grp
transgrp/data/trans24ca.grp
transgrp/data/trans24cb.grp
transgrp/data/trans24cc.grp
transgrp/data/trans24cd.grp
transgrp/data/trans24ce.grp
transgrp/data/trans24cf.grp
transgrp/data/trans24d.grp
transgrp/data/trans24e.grp
transgrp/data/trans24f.grp
transgrp/data/trans24g.grp
transgrp/data/trans24h.grp
transgrp/data/trans24i.grp
transgrp/data/trans24j.grp
transgrp/data/trans24k.grp
transgrp/data/trans24l.grp
transgrp/data/trans24m.grp
transgrp/data/trans24n.grp
transgrp/data/trans24o.grp
transgrp/data/trans24p.grp
transgrp/data/trans24q.grp
transgrp/data/trans24r.grp
transgrp/data/trans24s.grp
transgrp/data/trans24t.grp
transgrp/data/trans24u.grp
transgrp/data/trans24v.grp
transgrp/data/trans24w.grp
transgrp/data/trans24x.grp
transgrp/data/trans24y.grp
transgrp/data/trans24z.grp
transgrp/data/trans25.grp
transgrp/data/trans25a.grp
transgrp/data/trans26.grp
transgrp/data/trans26a.grp
transgrp/data/trans27.grp
transgrp/data/trans27a.grp
transgrp/data/trans27b.grp
transgrp/data/trans27c.grp
transgrp/data/trans27d.grp
transgrp/data/trans27e.grp
transgrp/data/trans27f.grp
transgrp/data/trans27g.grp
transgrp/data/trans27h.grp
transgrp/data/trans28.grp
transgrp/data/trans28a.grp
transgrp/data/trans28b.grp
transgrp/data/trans28c.grp
transgrp/data/trans28d.grp
transgrp/data/trans28e.grp
transgrp/data/trans28f.grp
transgrp/data/trans28g.grp
transgrp/data/trans29.grp
transgrp/data/trans30.grp
transgrp/data/trans30a.grp
transgrp/data/trans30b.grp
transgrp/data/trans30c.grp
transgrp/data/trans30d.grp
transgrp/data/trans30e.grp
transgrp/data/trans30f.grp
transgrp/data/trans30g.grp
transgrp/data/trans30h.grp
transgrp/data/trans30i.grp
transgrp/data/trans30j.grp
transgrp/data/trans30k.grp
transgrp/data/trans30l.grp
transgrp/data/trans30m.grp
transgrp/data/trans30n.grp
transgrp/data/trans30o.grp
transgrp/data/trans30p.grp
transgrp/data/trans30q.grp
transgrp/data/trans30r.grp
transgrp/data/trans30s.grp
transgrp/data/trans30t.grp
transgrp/data/trans8.grp
transgrp/data/trans9.grp
transgrp/doc/gapmacro.tex
transgrp/doc/manual.example-1.tst
transgrp/doc/._manual.tex
transgrp/doc/manual.tex
transgrp/doc/._transgrp.tex
transgrp/doc/transgrp.tex
transgrp/doc/tthmacros.tex
transgrp/htm/biblio.htm
transgrp/htm/CHAP001.htm
transgrp/htm/chapters.htm
transgrp/htm/indxA.htm
transgrp/htm/indxN.htm
transgrp/htm/indxO.htm
transgrp/htm/indxS.htm
transgrp/htm/indxT.htm
transgrp/htm/theindex.htm
transgrp/lib/trans.gd
transgrp/lib/trans.gi
transgrp/lib/trans.grp
transgrp/lib/trans10.grp
transgrp/lib/trans11.grp
transgrp/lib/trans12.grp
transgrp/lib/trans13.grp
transgrp/lib/trans14.grp
transgrp/lib/trans15.grp
transgrp/lib/trans16.grp
transgrp/lib/trans16a.grp
transgrp/lib/trans16b.grp
transgrp/lib/trans16c.grp
transgrp/lib/trans16d.grp
transgrp/lib/trans16e.grp
transgrp/lib/trans16f.grp
transgrp/lib/trans16g.grp
transgrp/lib/trans17.grp
transgrp/lib/trans18.grp
transgrp/lib/trans18a.grp
transgrp/lib/trans18b.grp
transgrp/lib/trans18c.grp
transgrp/lib/trans18d.grp
transgrp/lib/trans19.grp
transgrp/lib/trans20.grp
transgrp/lib/trans20a.grp
transgrp/lib/trans20b.grp
transgrp/lib/trans20c.grp
transgrp/lib/trans20d.grp
transgrp/lib/trans21.grp
transgrp/lib/trans21a.grp
transgrp/lib/trans22.grp
transgrp/lib/trans22a.grp
transgrp/lib/trans23.grp
transgrp/lib/trans24.grp
transgrp/lib/trans24a.grp
transgrp/lib/trans24aa.grp
transgrp/lib/trans24ab.grp
transgrp/lib/trans24ac.grp
transgrp/lib/trans24ad.grp
transgrp/lib/trans24ae.grp
transgrp/lib/trans24af.grp
transgrp/lib/trans24ag.grp
transgrp/lib/trans24ah.grp
transgrp/lib/trans24ai.grp
transgrp/lib/trans24aj.grp
transgrp/lib/trans24ak.grp
transgrp/lib/trans24al.grp
transgrp/lib/trans24am.grp
transgrp/lib/trans24an.grp
transgrp/lib/trans24ao.grp
transgrp/lib/trans24ap.grp
transgrp/lib/trans24aq.grp
transgrp/lib/trans24ar.grp
transgrp/lib/trans24as.grp
transgrp/lib/trans24at.grp
transgrp/lib/trans24au.grp
transgrp/lib/trans24av.grp
transgrp/lib/trans24aw.grp
transgrp/lib/trans24ax.grp
transgrp/lib/trans24ay.grp
transgrp/lib/trans24az.grp
transgrp/lib/trans24b.grp
transgrp/lib/trans24ba.grp
transgrp/lib/trans24bb.grp
transgrp/lib/trans24bc.grp
transgrp/lib/trans24bd.grp
transgrp/lib/trans24be.grp
transgrp/lib/trans24bf.grp
transgrp/lib/trans24bg.grp
transgrp/lib/trans24bh.grp
transgrp/lib/trans24bi.grp
transgrp/lib/trans24bj.grp
transgrp/lib/trans24bk.grp
transgrp/lib/trans24bl.grp
transgrp/lib/trans24bm.grp
transgrp/lib/trans24bn.grp
transgrp/lib/trans24bo.grp
transgrp/lib/trans24bp.grp
transgrp/lib/trans24bq.grp
transgrp/lib/trans24br.grp
transgrp/lib/trans24bs.grp
transgrp/lib/trans24bt.grp
transgrp/lib/trans24bu.grp
transgrp/lib/trans24bv.grp
transgrp/lib/trans24bw.grp
transgrp/lib/trans24bx.grp
transgrp/lib/trans24by.grp
transgrp/lib/trans24bz.grp
transgrp/lib/trans24c.grp
transgrp/lib/trans24ca.grp
transgrp/lib/trans24cb.grp
transgrp/lib/trans24cc.grp
transgrp/lib/trans24cd.grp
transgrp/lib/trans24ce.grp
transgrp/lib/trans24cf.grp
transgrp/lib/trans24d.grp
transgrp/lib/trans24e.grp
transgrp/lib/trans24f.grp
transgrp/lib/trans24g.grp
transgrp/lib/trans24h.grp
transgrp/lib/trans24i.grp
transgrp/lib/trans24j.grp
transgrp/lib/trans24k.grp
transgrp/lib/trans24l.grp
transgrp/lib/trans24m.grp
transgrp/lib/trans24n.grp
transgrp/lib/trans24o.grp
transgrp/lib/trans24p.grp
transgrp/lib/trans24q.grp
transgrp/lib/trans24r.grp
transgrp/lib/trans24s.grp
transgrp/lib/trans24t.grp
transgrp/lib/trans24u.grp
transgrp/lib/trans24v.grp
transgrp/lib/trans24w.grp
transgrp/lib/trans24x.grp
transgrp/lib/trans24y.grp
transgrp/lib/trans24z.grp
transgrp/lib/trans25.grp
transgrp/lib/trans25a.grp
transgrp/lib/trans26.grp
transgrp/lib/trans26a.grp
transgrp/lib/trans27.grp
transgrp/lib/trans27a.grp
transgrp/lib/trans27b.grp
transgrp/lib/trans27c.grp
transgrp/lib/trans27d.grp
transgrp/lib/trans27e.grp
transgrp/lib/trans27f.grp
transgrp/lib/trans27g.grp
transgrp/lib/trans27h.grp
transgrp/lib/trans28.grp
transgrp/lib/trans28a.grp
transgrp/lib/trans28b.grp
transgrp/lib/trans28c.grp
transgrp/lib/trans28d.grp
transgrp/lib/trans28e.grp
transgrp/lib/trans28f.grp
transgrp/lib/trans28g.grp
transgrp/lib/trans29.grp
transgrp/lib/trans30.grp
transgrp/lib/trans30a.grp
transgrp/lib/trans30b.grp
transgrp/lib/trans30c.grp
transgrp/lib/trans30d.grp
transgrp/lib/trans30e.grp
transgrp/lib/trans30f.grp
transgrp/lib/trans30g.grp
transgrp/lib/trans30h.grp
transgrp/lib/trans30i.grp
transgrp/lib/trans30j.grp
transgrp/lib/trans30k.grp
transgrp/lib/trans30l.grp
transgrp/lib/trans30m.grp
transgrp/lib/trans30n.grp
transgrp/lib/trans30o.grp
transgrp/lib/trans30p.grp
transgrp/lib/trans30q.grp
transgrp/lib/trans30r.grp
transgrp/lib/trans30s.grp
transgrp/lib/trans30t.grp
transgrp/lib/trans8.grp
transgrp/lib/trans9.grp

=====================end of the list of text files=======
=====================binary files========================
transgrp/.swp
transgrp/doc/manual.bbl
transgrp/doc/manual.idx
transgrp/doc/manual.lab
transgrp/doc/._manual.pdf
transgrp/doc/manual.pdf
transgrp/doc/manual.six
transgrp/doc/manual.toc
transgrp/doc/tthout

=====================end of the list of binary files=====
=====================ignored files=======================

=====================end of the list of ignored files====
```

**Comment by olexandr-konovalov** [comment]
I had a look at Jenkins tests. Certainly, because it's not loaded by default, there are diffs like below, which should disappear as soon as it will be made needed to run GAP:
```
                                    
########> Diff in /data/gap-jenkins/workspace/GAP-pkg-update-master-quicktest/\
GAPCOPTS/64build/GAPGMP/gmp/GAPTARGET/install/label/kovacs/GAP-pkg-update-mast\
er-snapshot/tst/testinstall/opers/StructureDescription.tst:10
# Input is:
List(AllTransitiveGroups(DegreeAction, 6), G -> StructureDescription(G:short))\
;
# Expected output:
[ "6", "S3", "D12", "A4", "3xS3", "2xA4", "S4", "S4", "S3xS3", "(3^2):4", 
  "2xS4", "A5", "(S3xS3):2", "S5", "A6", "S6" ]
# But found:
Error, Variable: 'AllTransitiveGroups' must have a value
########
```
and
```
########> Diff in /data/gap-jenkins/workspace/GAP-pkg-update-master-quicktest/\
GAPCOPTS/64build/GAPGMP/gmp/GAPTARGET/install/label/kovacs/GAP-pkg-update-mast\
er-snapshot/tst/testinstall/pgroups.tst:131
# Input is:
ForAll(List([1..11], i -> TransitiveGroup(8,i)), IsPGroup);
# Expected output:
true
# But found:
Error, TransitiveGroup: function is not yet defined
########
```
and
```
########> Diff in /data/gap-jenkins/workspace/GAP-pkg-update-master-quicktest/\
GAPCOPTS/64build/GAPGMP/gmp/GAPTARGET/install/label/kovacs/GAP-pkg-update-mast\
er-snapshot/tst/testinstall/stabchain.tst:19
# Input is:
List([2..7],
      x -> List([1..NrTransitiveGroups(x)],
        y -> TestGens(TransitiveGroup(x,y))));;
# Expected output:
# But found:
Error, NrTransitiveGroups: function is not yet defined
########
```

This overview may be also interesting:

```
### Checking variables in "transgrp", ver. 1.0
#I  Overwriting already installed help book 'matgrp not loaded'.
#I  Overwriting already installed help book 'matgrp not loaded'.
new global variables:
  CntOp( grp, orb, op )*
  NumBol( b )*
  PerfectSubgroupsAlternatingGroup( g, symconj )*
  SeqsOrbits( g, n )*
  SetsOrbits( g, n )*
  TRANSARRCACHE*
  TRANSCOMBCACHE*
  TRANSDIR*
  TransArrange( m, n )*
  TransCombinat( m, n )*

new methods:
  RepresentativesPerfectSubgroups( G )
    alternating
  RepresentativesPerfectSubgroups( G )
    symmetric
  TransitiveIdentification( ogrp )
    generic

other new globals (write protected):
  AllTransitiveGroups( arg )
  OneTransitiveGroup( arg )
  SelectTransitiveGroups( arglis, alle )*
  TRANSGrp( deg, nr )*
  TransGrpLoad( deg, nr )*

other new globals (not write protected):
  TRANSDEGREES*
  TRANSGRP*
  TRANSLENGTHS*
  TRANSNONDISCRIM*
  TRANSPARTNUM*
  TRANSPROPERTIES*
  TRANSSELECT*
  TRANSSIZES*
```

**Comment by olexandr-konovalov** [comment]
Version 2.0 picked up - now in testing.

**Comment by hulpke** [comment]
I just packed an updated version with the latest changes.

**Comment by olexandr-konovalov** [comment]
Thanks, but it does not pass validation:
```
* TransGrp - new version 2.0.1 discovered!!!
  ============================================
  Getting new archives from 
  http://www.math.colostate.edu/~hulpke/transgrp/transgrp2.0.1[ ".tar.gz" ]
  unpacking transgrp2.0.1.tar.gz
#E  component `Date' must be bound to a string of the form `dd/mm/yyyy'
  ERROR (TransGrp): validation of the info file not successful! SKIPPING!!!
```

In PackageInfo.g in
```
##  Release date of the current version in dd/mm/yyyy format.
Date := "3/11/2017",
```
please put 03/11/2017.

**Comment by hulpke** [comment]
I believe this is resolved now.
